From: Martin von Zweigbergk <martinvonz@google.com>
Date: Fri, 21 May 2021 12:03:33 -0400
Subject: tests: make test-archive.t pass on py3.9 (issue6504)

Something got stricter at parsing URL query parameters and now the
parameters need to be separated by "&"; ";" is no longer allowed. See
issue6504 for details.

Differential Revision: https://phab.mercurial-scm.org/D10472

Origin: upstream, https://www.mercurial-scm.org/repo/hg/rev/dc8976cc3a6e
Bug-Debian: https://bugs.debian.org/986514
Bug-upstream: https://bz.mercurial-scm.org/show_bug.cgi?id=6504
---
 tests/test-archive.t | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/test-archive.t b/tests/test-archive.t
index 606c9e2..384a04a 100644
--- a/tests/test-archive.t
+++ b/tests/test-archive.t
@@ -334,10 +334,10 @@ invalid arch type should give 404
   >     pass
   > if len(sys.argv) <= 3:
   >     node, archive = sys.argv[1:]
-  >     requeststr = 'cmd=archive;node=%s;type=%s' % (node, archive)
+  >     requeststr = 'cmd=archive&node=%s&type=%s' % (node, archive)
   > else:
   >     node, archive, file = sys.argv[1:]
-  >     requeststr = 'cmd=archive;node=%s;type=%s;file=%s' % (node, archive, file)
+  >     requeststr = 'cmd=archive&node=%s&type=%s&file=%s' % (node, archive, file)
   > try:
   >     stdout = sys.stdout.buffer
   > except AttributeError:
